tests/dist/mpi/examples/mpi_allgather.cpp

```diff
@@ -1,15 +1,9 @@
 #include <faabric/mpi/mpi.h>
```

Comment:
For all MPI examples I've just had to remove the `main` function, and change the method name and namespace.

Note that I've opted to add a nested namespace, `tests::mpi`, as some function names where quite general.